app/controllers/partials/user_cookies_partial_controller.rb

```diff
@@ -36,6 +36,7 @@ def set_journey_hint_by_status(idp_entity_id, status, rp_entity_id = nil)
     journey_hint_by_status_value["STATE"] = { IDP: idp_entity_id,
                                               RP: rp_entity_id.nil? ? session[:transaction_entity_id] : rp_entity_id,
                                               STATUS: status }
+    journey_hint_by_status_value["STATE"][:JOURNEY_TYPE] = session[:journey_type] if status == "PENDING"
```

Comment:
```suggestion
    journey_hint_by_status_value["STATE"][:VERIFY_JOURNEY_TYPE] = session[:journey_type] if status == "PENDING"
```
Maybe better naming?

Reply:
Yep, makes sense.